src/main/java/seedu/duke/Storage.java

```diff
@@ -0,0 +1,120 @@
+package seedu.duke;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+
+/**
+ * Manages task operation that involves underlying file storage.
+ */
+public class Storage {
+    private Path filePath;
+    private final static String delimiter = "\\|";
+
+    /**
+     * Constructs a new storage.
+     */
+    public Storage() {
+        this.filePath = openDataFile();
+    }
+
+    private Path openDataFile() {
+        Path filePath = null;
+        try {
+            String home = System.getProperty("user.home");
+            Path pathToDir = Paths.get(home, "Downloads", "data");
+
+            if (!Files.exists(pathToDir)) {
+                Files.createDirectories(pathToDir);
+            }
+            filePath = Paths.get(home, "Downloads", "data", "duke.txt");
+            if (!Files.exists(filePath)) {
+                Files.createFile(filePath);
+            }
+        } catch (IOException e) {
+            System.err.println(e.getMessage());
+        } finally {
+            return filePath;
+        }
+    }
+
+    private String[] readFromDataFile(int lineNumber) throws IOException {
+        String line = Files.readAllLines(filePath).get(lineNumber);
+        String[] lineArray = line.split(delimiter);
+        int len = lineArray.length;
+        for (int i = 0; i < len; i++) {
+            lineArray[i] = lineArray[i].trim();
+        }
+        return lineArray;
+    }
+
+    /**
+     * Converts the file content to a <code>TaskList</code> object.
+     *
+     * @return The TaskList converted from the file.
+     * @throws IOException If reading from the file fails.
+     */
+    public TaskList fileToArrayList() throws IOException {
+        int lineNumber = Math.toIntExact(Files.lines(filePath).count());
+        TaskList tasks = new TaskList(new ArrayList<>());
+        for (int i = 0; i < lineNumber; i++) {
+            String[] s = readFromDataFile(i);
```

Comment:
Maybe this is not a big issue, but a potential improvement could be using a more descriptive name, such as `fileString`, instead of `s`.